onnx/defs/reduction/defs.cc

```diff
@@ -35,38 +35,44 @@ False instead of True.)DOC";
         OpSchema::numeric_types_for_math_reduction(),
         "Constrain input and output types to high-precision numeric tensors.");
     schema.TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+      // Type inference
       propagateElemTypeFromInputToOutput(ctx, 0, 0);
+
+      // Shape inference
       if (!hasNInputShapes(ctx, 1)) {
         return;
       }
 
       int64_t keep_dims = 1;
-      auto attr_proto = ctx.getAttribute("keepdims");
-      if (attr_proto) {
+      const auto* attr_proto = ctx.getAttribute("keepdims");
+      if (attr_proto)
         keep_dims = attr_proto->i();
-      }
+
       auto& input_shape = ctx.getInputType(0)->tensor_type().shape();
       int64_t input_ndim = input_shape.dim_size();
       auto output_shape =
           ctx.getOutputType(0)->mutable_tensor_type()->mutable_shape();
       std::vector<int64_t> axes;
-      auto axes_proto = ctx.getAttribute("axes");
+      const auto* axes_proto = ctx.getAttribute("axes");
       if (axes_proto)
         axes.assign(axes_proto->ints().begin(), axes_proto->ints().end());
 
       for (size_t i = 0; i < axes.size(); ++i) {
         if (axes[i] < 0)
           axes[i] += input_ndim;
+        if (axes[i] < -input_ndim || 
```

Comment:
Is `axes[i] < -input_ndim` still strict after `axes[i] += input_ndim`? Not sure if you want to use
`axes[i] >= 0` in Line 63.